toolchain/sem_ir/inst.h

```diff
@@ -305,38 +305,11 @@ struct LocIdAndInst {
   template <typename InstT>
     requires(!Internal::HasNodeId<InstT>)
   static auto NoLoc(InstT inst) -> LocIdAndInst {
-    return LocIdAndInst(LocId::Invalid, inst, /*is_untyped=*/true);
+    return {.loc_id = LocId::Invalid, .inst = inst};
   }
 
-  // For the common case, support construction as:
-  //   context.AddInst({node_id, SemIR::MyInst{...}});
-  template <typename InstT>
-    requires(Internal::HasNodeId<InstT>)
-  LocIdAndInst(decltype(InstT::Kind)::TypedNodeId node_id, InstT inst)
-      : loc_id(node_id), inst(inst) {}
-
-  // If TypedNodeId is Parse::NodeId, allow construction with a LocId.
-  // TODO: This is somewhat historical due to fetching the NodeId from insts()
-  // for things like Temporary; should we require Untyped in these cases?
-  template <typename InstT>
-    requires(std::same_as<typename decltype(InstT::Kind)::TypedNodeId,
-                          Parse::NodeId>)
-  LocIdAndInst(LocId loc_id, InstT inst) : loc_id(loc_id), inst(inst) {}
```

Comment:
If I'm reading this right: it looks like we no longer have any compile-time enforcement that the node kind associated with an instruction matches the node kind specified in the instruction definition. Is that correct? That seems unfortunate; is it necessary?

Reply:
Sure, I can do that approach instead. Restored.